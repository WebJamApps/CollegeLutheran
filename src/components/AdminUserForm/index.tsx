import React from 'react';
import type { AdminDashboard } from '../../containers/AdminDashboard';

type PageProps = {comp:AdminDashboard};
const AdminUserForm = ({ comp }:PageProps): JSX.Element => {
  const { formError } = comp.state;
  return (
    <div
      className="material-content elevation3"
      style={{ maxWidth: '320px', margin: '30px auto', padding: '10px 10px 20px 10px' }}
    >
      <h4 className="material-header-h4">
        Add Admin User
      </h4>
      <form
        id="modify-admins"
        style={{
          textAlign: 'left', marginLeft: '4px', width: '100%', maxWidth: '100%',
        }}
      >
        <label htmlFor="addAdminEmail">
          Admin Email
          <input
            id="addAdminEmail"
            type="email"
            placeholder="placeholder@gmail.com"
            onChange={comp.onChangeAdminEmail}
          />
        </label>
        <input type="submit" disabled={comp.controller.validateAdmin()} onClick={comp.controller.addAdminUser} />
        <p className="form-errors" style={{ color: 'red', marginBottom: '-15px' }}>{formError}</p>
      </form>
    </div>
  );
};

export default AdminUserForm;
