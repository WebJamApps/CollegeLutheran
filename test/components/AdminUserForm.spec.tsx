/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import AdminUserForm from '../../src/components/AdminUserForm';

describe('AdminUserForm', () => {
  let wrapper:ShallowWrapper, adminDashboardStub:any = {};
  beforeEach(() => {
    adminDashboardStub = {
      onChangeAdminEmail: jest.fn(),
      state: { formError: 'bad' },
      controller: { validateAdmin: jest.fn() },
    };
    wrapper = shallow(<AdminUserForm comp={adminDashboardStub} />);
  });
  it('renders correctly', () => { expect(wrapper).toMatchSnapshot(); });
});
