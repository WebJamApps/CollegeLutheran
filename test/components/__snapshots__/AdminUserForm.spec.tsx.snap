// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminUserForm renders correctly 1`] = `ShallowWrapper {}`;
